.changeset/mean-horses-kiss.md

```diff
@@ -0,0 +1,26 @@
+---
+'astro': minor
+---
+
+Adds a new `injectTypes` utility to the Integration API and refactors how typegen works
+
+Integrations can now use a new `injectTypes` utility exposed on the `astro:config:done` hook. It allows to inject types easily in the user's project. `filename` must ends with `".d.ts"` and `content` must be valid TypeScript (it will be formatted).
+
+Under the hood, it will create a file at `/.astro/integrations/<normalized_integration_name>/<normalized_filename>.d.ts` and create references to it. `injectTypes` returns a URL to the normalized path.
```

Comment:
Suggesting a couple tweaks here. In general in docs we show a `()` after function names to distinguish them from non-function variables:

```suggestion
Adds a new `injectTypes()` utility to the Integration API and refactors how type generation works

Integrations can now use a new `injectTypes()` utility in the `astro:config:done` hook. This utility allows an integration to inject types into a user's project. `filename` must end with `".d.ts"` and `content` must be valid TypeScript (it will be formatted).

Under the hood, it will create a file at `/.astro/integrations/<normalized_integration_name>/<normalized_filename>.d.ts` and create references to it. `injectTypes()` returns a URL to the normalized path.
```